### ISSUE TITLE
ImportError: cannot import name canonical_filename

### ISSUE THREAD

**Issue by adaszko** [HypothesisWorks/hypothesis#1378]
Hi, I'm getting an import error on startup:

```
  File "/Users/adaszko/repos/fieldaware/fieldaware-venv/lib/python2.7/site-packages/hypothesis/core.py", line 38, in <module>
    from coverage.files import canonical_filename
ImportError: cannot import name canonical_filename
```

I've downloaded https://files.pythonhosted.org/packages/4b/e4/5ebf3220993de03f2120a16d9e91cfd053f4c11ada0cf033f2bfe9683fcf/hypothesis-3.65.0-py2-none-any.whl and the `METADATA` file there specifies dependency on `coverage` without any version number:

```
% grep coverage METADATA
Requires-Dist: coverage
```

My local `coverage` is at `3.7.1`.  It works if I upgrade `coverage` to `4.4.1`, so I think there's an issue in hypothesis in that it doesn't specify the version bound on `coverage`.

**Comment by alexwlchan** [comment]
Thanks for the report!

Yep, we don’t specify a version of coverage in our `setup.py`.

I think this is all the imports from coverage we use:

```python
from coverage import Coverage, CoverageData
from coverage.collector import Collector, FileDisposition
from coverage.files import canonical_filename
from coverage.tracer import CFileDisposition as FileDisposition
import coverage
```

I’ll work out what the minimum version we need is, and add a bound in `setup.py`.

**Comment by alexwlchan** [comment]
Okay, so we want v4.0 – and not any of the alpha versions! Fix available in #1379.

**Comment by adaszko** [comment]
Great, thanks!
